internal/ls/types.go

```diff
@@ -15,3 +15,36 @@ type Location struct {
 	FileName string
 	Range    core.TextRange
 }
+
+type JsxAttributeCompletionStyle string
+
+const (
+	JsxAttributeCompletionStyleAuto   JsxAttributeCompletionStyle = "auto"
+	JsxAttributeCompletionStyleBraces JsxAttributeCompletionStyle = "braces"
+	JsxAttributeCompletionStyleNone   JsxAttributeCompletionStyle = "none"
+)
+
+type UserPreferences struct {
+	// Enables auto-import-style completions on partially-typed import statements. E.g., allows
+	// `import write|` to be completed to `import { writeFile } from "fs"`.
+	IncludeCompletionsForImportStatements *bool
```

Comment:
Should these just be Tristate?

Reply:
They could be. To answer this and your other comment above, I haven't looked at how to handle those custom options yet via LSP. I left things as `*bool` because that's closer to what we get from the LSP generated types.